logging/api/LoggingTest/LoggingTest.cs

```diff
@@ -0,0 +1,239 @@
+﻿/*
+ * Copyright (c) 2016 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+using System;
+using Google.Logging.V2;
+using Xunit;
+using System.IO;
+using System.Diagnostics;
+using System.Threading;
+
+namespace GoogleCloudSamples
+{
+    public class BaseTest
+    {
+        private readonly string _projectId;
+
+        public BaseTest()
+        {
+            _projectId = Environment.GetEnvironmentVariable("GOOGLE_PROJECT_ID");
+        }
+
+        public struct ConsoleOutput
+        {
+            public int ExitCode;
+            public string Stdout;
+        };
+
+        /// <summary>Runs LoggingSample.exe with the provided arguments</summary>
+        /// <returns>The console output of this program</returns>
+        public static ConsoleOutput Run(params string[] arguments)
+        {
+            Console.Write("LoggingSample.exe ");
+            Console.WriteLine(string.Join(" ", arguments));
+
+            using (var output = new StringWriter())
+            {
+                LoggingSample loggingSample = new LoggingSample(output);
+                var consoleOutput = new ConsoleOutput()
+                {
+                    ExitCode = loggingSample.Run(arguments),
+                    Stdout = output.ToString()
+                };
+                Console.Write(consoleOutput.Stdout);
+                return consoleOutput;
+            }
+        }
+
+        protected static void AssertSucceeded(ConsoleOutput output)
+        {
+            Assert.True(0 == output.ExitCode,
+                $"Exit code: {output.ExitCode}\n{output.Stdout}");
+        }
+
+        public class LoggingTest : BaseTest, IDisposable
+        {
+            public void Dispose() { }

```

Comment:
If Dispose() is empty, then we can just drop it and stop implementing IDisposable.